### PR TITLE
Fix S1123 FN: The explanation should not be null or whitespace

### PR DESCRIPTION
**Pull request by Corniel**
As mentioned by @martin-strecker-sonarsource reading the arguments should be improved. Result of a remark in #6593 .

### CODE REVIEW

**Comment by Corniel** [comment]
@martin-strecker-sonarsource Given #6593 this should be an easy one, right?

File: analyzers/src/SonarAnalyzer.VisualBasic/Rules/ObsoleteAttributes.cs [hunk 1]
```diff
@@ -24,4 +24,10 @@ namespace SonarAnalyzer.Rules.VisualBasic;
 public sealed class ObsoleteAttributes : ObsoleteAttributesBase<SyntaxKind>
 {
     protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
+
+    protected override SyntaxNode GetExplanationExpression(SyntaxNode node) =>
+       node is AttributeSyntax { ArgumentList.Arguments: { Count: >= 1 } arguments }
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.VisualBasic/Rules/ObsoleteAttributes.cs:29]: You don't need to fix this as it should be done as part of the rework of the ParameterLookup, but for Obsolete it is not as easy:
```cs
    [Obsolete(error: true, message: "Test")]
```
We can live with the false negative here for the moment.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ObsoleteAttributesNeedExplanation.cs [hunk 2]
```diff
@@ -12,6 +12,15 @@ class Noncompliant
...
 
+    [Obsolete(null)] // Noncompliant
+    void WithNull() { }
+
+    [Obsolete("")] // Noncompliant
+    void WithEmptyString() { }
+
+    [Obsolete("  ")] // Noncompliant
+    void WithWhiteSpace() { }
+
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ObsoleteAttributesNeedExplanation.cs:23]: Please add more test cases:
```cs
    [Obsolete("", true)] // Noncompliant
    [Obsolete(error: true, message: "")] // FN
    [Obsolete("", true)] // Noncompliant
    [Obsolete("", true, DiagnosticId = "123")] // Noncompliant
    [Obsolete("", true, DiagnosticId = "123", UrlFormat ="http://docs.domain.com")] // Noncompliant
    [Obsolete("", DiagnosticId = "123", UrlFormat ="http://docs.domain.com")] // Noncompliant
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ObsoleteAttributesNeedExplanation.vb [hunk 3]
```diff
@@ -14,6 +14,18 @@ Class Noncompliant
...
+    End Sub
+
+    <Obsolete("")> ' Noncompliant
+    Sub WithEmptyString() { }
+    End Sub
+
+    <Obsolete("  ")> ' Noncompliant
+    Sub WithWhiteSpace() { }
+    End Sub
+
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ObsoleteAttributesNeedExplanation.vb:28]: Here too like so.
```vb
<Obsolete("", True)>
<Obsolete("", True, DiagnosticId:="123")>
<Obsolete("", True, DiagnosticId:="123", UrlFormat:="http://docs.domain.com")>
<Obsolete("", DiagnosticId:="123", UrlFormat:="http://docs.domain.com")>
```

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
Please add some more test cases

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ObsoleteAttributesNeedExplanation.vb (hunk 3 above)
Comment (Corniel) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ObsoleteAttributesNeedExplanation.vb:28]: Added extra testcases, and scenario's resulting in FP, FN, and one resulting in a noncompliant for the wrong reason.

**Comment by Corniel** [comment]
@martin-strecker-sonarsource Moved some test cases to tests that now only run under `#if NET` (I forgot the run the tests locally under .NET 4.8, oops). I used language versions to achieve this, although is in the end not language but framework dependent.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/ObsoleteAttributesTest.cs [hunk 4]
```diff
@@ -56,6 +56,10 @@ public ObsoleteAttributesTest()
     [TestMethod]
     public void ObsoleteAttributesNeedExplanation_CSharp10() =>
         explanationNeededCS.AddPaths("ObsoleteAttributesNeedExplanation.CSharp10.cs").WithOptions(ParseOptionsHelper.FromCSharp10).Verify();
+
+    [TestMethod]
+    public void ObsoleteAttributesNeedExplanation_VB14() =>
+        explanationNeededVB.AddPaths("ObsoleteAttributesNeedExplanation.VB14.vb").WithOptions(ParseOptionsHelper.FromVisualBasic14).Verify();
 #endif
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/ObsoleteAttributesTest.cs:63]:
```suggestion

#endif
```

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**